src/internal/connector/exchange/service_restore.go

```diff
@@ -283,6 +283,19 @@ func SendMailToBackStore(
 
 	for _, attachment := range attached {
 		if err := uploadAttachment(ctx, uploader, attachment); err != nil {
+			if *attachment.GetOdataType() == "#microsoft.graph.itemAttachment" {
```

Comment:
I'm possibly being paranoid here - but a `attachment != nil` check here will be good.